lib/colocated-babel-plugin.js

```diff
@@ -22,7 +22,15 @@ module.exports = function (babel) {
       },
 
       ExportDefaultDeclaration(path, state) {
-        if (state.colocatedTemplateFound !== true || state.setComponentTemplateInjected === true) {
+        let defaultExportDeclarationPath = path.get('declaration');
+        let defaultExportIsExpressionOrClass =
+          defaultExportDeclarationPath.isClass() || defaultExportDeclarationPath.isExpression();
```

Comment:
I don't think this allows for all the things that we need to allow for.

Specifically, it should be possible to do:

```js
export default function myThing() {}
setComponentManager((owner) => new SomeManager(owner), myThing);
```

[Demo here](https://astexplorer.net/#/gist/ac530a4a422fbbe9a92328ca51165e3f/33a633d5291489e5f3f6690daf385107625fdae1)

I think adding `isFunction` would work to address this.

```suggestion
        let defaultExportIsExpressionOrClass =
          defaultExportDeclarationPath.isClass() || defaultExportDeclarationPath.isExpression() || defaultExportDeclarationPath.isFunction();
```

Reply:
You're right, thanks! I forgot named functions act as declarations rather than expressions too